omnipaxos/src/ballot_leader_election.rs

```diff
@@ -32,32 +34,31 @@ impl Ballot {
     /// # Arguments
     /// * `n` - Ballot number.
     /// * `pid` -  Used as tiebreaker for total ordering of ballots.
-    pub fn with(n: u32, pid: NodeId) -> Ballot {
-        Ballot { n, pid }
+    pub fn with(n: u32, priority: u32, pid: NodeId) -> Ballot {
+        Ballot { n, priority, pid }
     }
 }
 
 /// Used to define a BLE epoch
 #[derive(Clone, Copy, Eq, Debug, Default, PartialEq)]
-struct BLEBallot {
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct BLEBallot {
     /// Ballot of a replica.
-    ballot: Ballot,
-    /// Custom priority parameter
-    priority: u32,
+    pub ballot: Ballot,
     /// Used to determine if the replica is a candidate to become a leader or remain a leader.
-    connectivity: u8,
+    pub connectivity: u8,
```

Comment:
This struct is nice for code readability i.e. having ballots: `Vec<BLEBallot>` instead of `Vec<(Ballot, u8)`. But I prefer not to have it as a public struct as that might confuse users (with `Ballot)` in the API docs. I think we should have it as a private struct and send `Ballot` and `connectivity` as separate fields in `HeartbeatReply`. We can then construct the BLEBallot before we add it to `self.ballots`